src/data-fetcher/bigwig/higlass-bigwig-datafetcher.ts

```diff
@@ -0,0 +1,323 @@
+/* TODO: Support esm
+ * This code is based on the following repo:
+ * https://github.com/higlass/higlass-bigwig-datafetcher/blob/main/src/BigwigDataFetcher.js
+ */
+import { BigWig } from '@gmod/bbi';
+import type { Assembly } from '@gosling.schema';
+import { RemoteFile } from 'generic-filehandle';
+import { GET_CHROM_SIZES } from '../../core/utils/assembly';
+
+import type { Feature } from '@gmod/bbi';
+import type { ChromInfo, TilesetInfo } from '@higlass/types';
+
+type DataConfig = {
+    url: string;
+    chromSizesUrl: string;
+};
+
+type Tile = {
+    tilePos: [number];
+    tileId: string;
+    zoomLevel: number;
+    min_value: number;
+    max_value: number;
+    dense: (number | null)[];
+    denseDataExtrema: InstanceType<import('@higlass/types').HGC['utils']['DenseDataExtrema1D']>;
+    minNonZero: number;
+    maxNonZero: number;
+};
+
+type BigWigHeader = {
+    zoomLevels: { reductionLevel: number }[];
+};
+
+type ExtendedFeature = Feature & { startAbs: number; endAbs: number };
+
+function BBIDataFetcher(HGC: import('@higlass/types').HGC, dataConfig: DataConfig) {
+    if (!new.target) {
+        throw new Error('Uncaught TypeError: Class constructor cannot be invoked without "new"');
+    }
+
+    const cls = class BBIDataFetcherClass {
+        dataConfig: typeof dataConfig;
+        bwFileHeader: BigWigHeader | null;
+        bwFile: BigWig | null;
+        TILE_SIZE: number;
+        errorTxt: string;
+        dataPromises: Promise<unknown>[];
+        chromSizes: ChromInfo<string> & { chrToAbs: (name: string, pos: number) => number };
+        assembly?: Assembly;
+        tilesetInfoLoading?: boolean;
+
+        constructor() {
+            // TODO(2022-06-29): shouldn't have hard coded chromSizesUrl
```

Comment:
I should have dealt with the assembly issue with much care. I will address this when I work on supporting custom assembly #582 soon

Reply:
great!